mainnet-contracts/src/PufferVaultV3.sol

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     * @notice Returns the total assets held by the vault.
+     * @dev Returns the total assets held by the vault, including ETH held in the eigenpods as a result of receiving rewards.
+     * @return The total assets held by the vault.
+     */
+    function totalAssets() public view virtual override returns (uint256) {
+        return (super.totalAssets() + getTotalRewardMintAmount());
+    }
+
+    /**
+     * @notice Returns the total reward mint amount.
+     * @return The total reward mint amount.
+     */
+    function getTotalRewardMintAmount() internal view returns (uint256) {
+        VaultStorage storage $ = _getPufferVaultStorage();
+        return $.totalRewardMintAmount;
+    }
+
+    /**
+     * @notice Mints and bridges rewards according to the provided parameters.
+     * @param params The parameters for bridging rewards.
+     */
+    function mintAndBridgeRewards(MintAndBridgeParams calldata params) external payable restricted {
+        VaultStorage storage $ = _getPufferVaultStorage();
+
+        if (params.rewardsAmount > $.allowedRewardMintAmount) {
+            revert InvalidMintAmount();
+        }
+
+        if (($.lastRewardMintTimestamp + $.allowedRewardMintFrequency) > block.timestamp) {
+            revert NotAllowedMintFrequency();
+        }
+
+        BridgeData memory bridgeData = $.bridges[params.bridge];
+
+        if (bridgeData.l2RewardManager == address(0)) {
+            revert BridgeNotAllowlisted();
+        }
+
+        uint256 ethToPufETHRate = previewDeposit(1 ether);
+        uint256 shares = ethToPufETHRate.mulDiv(params.rewardsAmount, 1 ether, Math.Rounding.Floor);
+
+        $.lastRewardMintTimestamp = uint40(block.timestamp);
+        $.totalRewardMintAmount += uint104(params.rewardsAmount);
+
+        _mint(address(this), shares);
+        _approve(address(this), address(LOCKBOX), params.rewardsAmount);
+        LOCKBOX.deposit(shares);
+
+        // This contract approves transfer to the bridge
+        XTOKEN.approve(address(params.bridge), shares);
+
+        MintAndBridgeCalldata memory bridgingCalldata = MintAndBridgeCalldata({
+            rewardsAmount: uint128(params.rewardsAmount),
+            ethToPufETHRate: uint128(ethToPufETHRate),
+            startEpoch: params.startEpoch,
+            endEpoch: params.endEpoch,
+            rewardsRoot: params.rewardsRoot,
+            rewardsURI: params.rewardsURI
+        });
+
+        BridgingParams memory bridgingParams =
+            BridgingParams({ bridgingType: BridgingType.MintAndBridge, data: abi.encode(bridgingCalldata) });
+
+        // Encode calldata for the target contract call
+        bytes memory encodedCallData = abi.encode(bridgingParams);
+
+        IBridgeInterface(params.bridge).xcall{ value: msg.value }({
+            destination: bridgeData.destinationDomainId, // Domain ID of the destination chain
+            to: bridgeData.l2RewardManager, // Address of the target contract
```

Comment:
Shouldn't this be a constant? On L2, we are using Upgradeable contract. This can be an immutable